pkg/controller/kogitoinfra/infinispan.go

```diff
@@ -119,63 +119,82 @@ func (i *infinispanInfraReconciler) getInfinispanSecretEnvVars(infinispanInstanc
 	envProps = append(envProps, framework.CreateEnvVar(infinispanEnablePersistenceEnvKey, "true"))
 	secretName := customInfinispanSecret.Name
 	envProps = append(envProps, framework.CreateEnvVar(infinispanEnvKeyCredSecret, secretName))
-	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesInfinispanSpring[envVarInfinispanUser], secretName, infrastructure.InfinispanSecretUsernameKey))
-	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesInfinispanQuarkus[envVarInfinispanUser], secretName, infrastructure.InfinispanSecretUsernameKey))
-	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesInfinispanSpring[envVarInfinispanPassword], secretName, infrastructure.InfinispanSecretPasswordKey))
-	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesInfinispanQuarkus[envVarInfinispanPassword], secretName, infrastructure.InfinispanSecretPasswordKey))
+	if runtime == v1beta1.QuarkusRuntimeType {
+		envProps = append(envProps, framework.CreateSecretEnvVar(propertiesInfinispanQuarkus[envVarInfinispanUser], secretName, infrastructure.InfinispanSecretUsernameKey))
+		envProps = append(envProps, framework.CreateSecretEnvVar(propertiesInfinispanQuarkus[envVarInfinispanPassword], secretName, infrastructure.InfinispanSecretPasswordKey))
+	} else if runtime == v1beta1.SpringBootRuntimeType {
+		envProps = append(envProps, framework.CreateSecretEnvVar(propertiesInfinispanSpring[envVarInfinispanUser], secretName, infrastructure.InfinispanSecretUsernameKey))
+		envProps = append(envProps, framework.CreateSecretEnvVar(propertiesInfinispanSpring[envVarInfinispanPassword], secretName, infrastructure.InfinispanSecretPasswordKey))
+	}
```

Comment:
Hey Kevin!

Alternatively, you could do something like:

```go
envProps = append(envProps, framework.CreateSecretEnvVar(properties[runtime][envVarInfinispanUser], secretName, infrastructure.InfinispanSecretUsernameKey))
envProps = append(envProps, framework.CreateSecretEnvVar(properties[runtime][envVarInfinispanPassword], secretName, infrastructure.InfinispanSecretPasswordKey))
```

`properties` could be a map with the key being the target runtime and the value a map with the related properties.

The code as you proposed, is pretty much the same with added conditionals.